http-verbs-common/src/main/scala/uk/gov/hmrc/http/play/HttpClient2.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// TODO putting this in this package means that all clients which do
+// `import uk.gov.hmrc.http._` will then have to make play imports with _root_ `import _root_.play...`
+package uk.gov.hmrc.http.play
+
+import play.api.libs.ws.{BodyWritable, WSRequest}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+
+import java.net.URL
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.runtime.universe.TypeTag
+
+trait HttpClient2 {
```

Comment:
This is the new interface for clients. Note, it can provide proxy per endpoint (BDOG-1421), replace user-agent, request timeouts on demand etc